include/nlohmann/detail/macro_scope.hpp

```diff
@@ -106,6 +106,16 @@
     #endif
 #endif
 
+// see https://devblogs.microsoft.com/cppblog/msvc-cpp20-and-the-std-cpp20-switch/
+// #if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 192829913
+//     #define JSON_NO_UNIQUE_ADDRESS [[msvc::no_unique_address]]
+// #el
+#if JSON_HEDLEY_HAS_ATTRIBUTE(no_unique_address)
+    #define JSON_NO_UNIQUE_ADDRESS [[no_unique_address]]
+#else
+    #define JSON_NO_UNIQUE_ADDRESS
+#endif
```

Comment:
Still don't know if it's worth adding on MSVC. Long-term we may want to add a macro for ABI stability. Then, if users opt-out of ABI stability, we could enable the optimization on MSVC. But that would only be worth the effort if it could be used elsewhere.

Reply:
Regarding ABI stability, I am torn. On the one hand, it's nice to have and all. On the other hand, it's so hard to check and brings so many limitations that it could start to become a blocker for a header-only library...

Reply:
I'm leaning towards removing `[[msvc::no_unique_address]]` but keeping `[[no_unique_address]]`. It's a fairly minor optimization.

Reply:
What is the issue with `[[msvc::no_unique_address]]`?

Reply:
It's enabled regardless of C++ standard version and might catch users off guard. In terms of ABI stability which would otherwise not be affected.

Reply:
Then I agree. I think it's one homework to decide to which extend ABI stability is something to strive for.